producer-c/producer-cloudwatch-integ/canary/CanaryLogsUtils.cpp

```diff
@@ -5,6 +5,7 @@
 #include "CanaryUtils.h"
 
 PCloudwatchLogsObject gCloudwatchLogsObject = NULL;
+std::mutex logLock; // Protect access to gCloudwatchLogsObject
```

Comment:
I really don't like globals here at all. Is there a way to avoid them?

Reply:
gLogLock?

Reply:
Not a fan of the global either. But for the current design, this is the only way. I am looking to revamp the canary at some point to move to a class based implementation. That will avoid use of globals altogether